### PR TITLE
Only add non-existent files to recent files if CheckFilesExist is false

### LINKED ISSUE #63: Check files exist setting has no effect
Non-existent files should be filtered out of the listing of recent files when this is set.

### PR DESCRIPTION
**Pull request by mikkelbu**
Fixes #63

### CODE REVIEW

**Comment by mikkelbu** [comment]
@CharliePoole I'm able to merge the PR and also dismiss the review, but I think I'll add some tests.
image

I'm not sure about changing the logic as we need to keep track on the index in both collections, but I'm open for suggestions (e.g. we could replace part of the check in the if-statement with a `if (....) break` inside the loop, but seems more complicated).

File: src/TestCentric/testcentric.gui/RecentFileMenuHandler.cs [hunk 1]
```diff
@@ -72,13 +75,15 @@ public void Load()
...
-                    // The V2 GUI doesn't show non-existent files, but we do
-                    MenuItem item = new MenuItem(String.Format("{0} {1}", index++, entry));
-                    item.Click += new System.EventHandler(OnRecentFileClick);
-                    Menu.MenuItems.Add(item);
+                    if (!checkFilesExist || File.Exists(entry))
+                    {
+                        MenuItem item = new MenuItem(String.Format("{0} {1}", index++, entry));
+                        item.Click += new EventHandler(OnRecentFileClick);
+                        Menu.MenuItems.Add(item);
+                    }
```
[the remark this answers is not in the document]
Comment (mikkelbu) [reply src/TestCentric/testcentric.gui/RecentFileMenuHandler.cs]: I can live with that 😄 
I started adding tests and discovered that we already have a testfixture for RecentFileMenuHandler.cs (in https://github.com/TestCentric/testcentric-gui/blob/master/src/TestCentric/tests/RecentFileMenuHandlerTests.cs), but the file was not included in any project, so I need to spent some time making the file compile (and the test succeed). Hopefully, I will have time to look at this tonight.

**Comment by mikkelbu** [comment]
I've simplified the logic a bit and also reintroduce the file `RecentFileMenuHandlerTests` and added tests of the new functionality. The mocking of the `UserSettings` class could be prettier, but that would require that we introduce interfaces for all settings-classes, and I think that would be too much for this PR.

**Comment by CharliePoole** [comment]
Looks good to me. I have a local branch with a pretty extensive set of fake implementations for the engine, including RecentFiles. Once this is merged, I'll rebase and then change these tests to use the fake handler version that I added to test utilities.

**Comment by immeraufdemhund** [comment]
Sorry about the delay on that.

**Comment by mikkelbu** [comment]
No need to be sorry. We are all doing this in our spare time, so delays should be expected 😄.

**Pull request merged**